tests/acceptance/commercial/test_delta_update_module.py

```diff
@@ -51,3 +52,77 @@ def test_build_and_run_module(
         assert "mender-binary-delta" in [
             line.split("/")[5] for line in output.split("\n") if line.startswith("/")
         ]
+
+    @pytest.mark.only_with_image("ext4")
+    def test_runtime_checksum(
+        self,
+        setup_board,
+        prepared_test_build,
+        bitbake_variables,
+        bitbake_image,
+        connection,
+        http_server,
+        board_type,
+        use_s3,
+        s3_address,
+    ):
+        """Check that the checksum of the running root filesystem is what we
+        expect. This is important in order for it to match when applying a delta
+        update.
+
+        """
+
+        if (
+            "read-only-rootfs"
+            not in bitbake_variables["IMAGE_FEATURES"].strip().split()
+        ):
+            pytest.skip("Only works when using read-only-rootfs IMAGE_FEATURE")
+
+        build_image(
+            prepared_test_build["build_dir"],
+            prepared_test_build["bitbake_corebase"],
+            bitbake_image,
+            ['IMAGE_INSTALL_append = " mender-binary-delta"'],
+            [
+                'BBLAYERS_append = " %s/../meta-mender-commercial"'
+                % bitbake_variables["LAYERDIR_MENDER"]
+            ],
+        )
+
+        image = latest_build_artifact(
+            prepared_test_build["build_dir"], "core-image*.mender"
+        )
+
+        try:
+            os.symlink(image, "temp-artifact.mender")
```

Comment:
Why do we do this, can't we install `image`? It has no implications, but after the update we will be doing reading back the original image (instead of the symlink).

Reply:
Okey, I guess is because of this comment in `install_update`:
```
# Note: `image` needs to be in current directory.
```
A cleaner solution can be to do this symlink inside the helper method.

Reply:
Good idea, I'll try that.

Reply:
Since I didn't want to modify the code that I moved in the same commit, and I had already made a merge with master in between this PR and the review, I decided to merge this and make a followup PR instead: https://github.com/mendersoftware/meta-mender/pull/1004